### PR TITLE
refactor: mv `ChangedAccount`

### PR DESCRIPTION
**Pull request by tcoratger**
Before the `ChangedAccount` structure was included in the transaction pool crate. However, as it is used mainly with our execution types, it seems more straightforward to move it to our execution type crate as long as the `changed_accounts` method which is now directly implemented for `ExecutionOutcome`.

### CODE REVIEW

**Review by onbjerg — APPROVED** [review]
lgtm

File: crates/evm/execution-types/src/execution_outcome.rs [hunk 1]
```diff
@@ -9,6 +9,24 @@ use revm::{
 };
 use std::collections::HashMap;
 
+/// Represents a changed account
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Default)]
```
Comment (rkrasiuk) [inline crates/evm/execution-types/src/execution_outcome.rs:13]: should not derive default 
```suggestion
#[derive(Clone, Copy, Debug, PartialEq, Eq)]
```

**Comment by onbjerg** [comment]
tests are failing somehow, can you take a look?

**Comment by tcoratger** [comment]
> tests are failing somehow, can you take a look?

@onbjerg Just fixed

**Review by mattsse — APPROVED** [review]
makes sense, interestingly we don't have this particular account type yet

**Pull request merged**